app/src/main/java/org/mozilla/fenix/crashes/CrashReporterController.kt

```diff
@@ -4,81 +4,78 @@
 
 package org.mozilla.fenix.crashes
 
-import android.util.Log
-import androidx.navigation.NavController
+import androidx.annotation.VisibleForTesting
 import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
-import mozilla.components.lib.crash.Crash
-import org.mozilla.fenix.R
+import org.mozilla.fenix.components.AppStore
 import org.mozilla.fenix.components.Components
-import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.components.appstate.AppAction
 import org.mozilla.fenix.utils.Settings
 
 class CrashReporterController(
-    private val crash: Crash,
-    private val sessionId: String?,
-    private val navController: NavController,
-    private val components: Components,
-    private val settings: Settings
+    @get:VisibleForTesting internal val sessionId: String,
+    @get:VisibleForTesting internal val components: Components,
+    @get:VisibleForTesting internal val settings: Settings,
+    @get:VisibleForTesting internal val appStore: AppStore
```

Comment:
Wondering how come we had to make all of these `@get:VisibleForTesting internal`

Reply:
`VisibleForTesting(otherwise = PRIVATE) internal` is the general approach used to be able to access otherwise private functions and properties in tests.
These properties are used in `CrashContentIntegrationTest` to test that the controller is constructed with the proper arguments.